src/reporters/http_sender.js

```diff
@@ -103,6 +103,7 @@ export default class HTTPSender {
 
     const result = this._jaegerThrift.Batch.rw.toBuffer(this._batch);
     if (result.err) {
+      this._reset();
```

Comment:
I think it's better to move this to the next line after toBuffer() call, and remove the same statement  from L111.